### PR TITLE
Add aliases to ASTCodegen for dmd#8591

### PR DESCRIPTION
**Pull request by thewilsonator**
So that #8591's dtoh can be built against an unmodified dmd (e.g as a dub package).

cc @ibuclaw

### CODE REVIEW

**Review by jacob-carlborg — APPROVED** [review]

File: src/dmd/astcodegen.d [hunk 1]
```diff
@@ -41,12 +41,31 @@ struct ASTCodegen
 
     alias MODFlags                  = dmd.mtype.MODFlags;
     alias Type                      = dmd.mtype.Type;
-    alias Tident                    = dmd.mtype.Tident;
```
Comment (ZombineDev) [inline src/dmd/astcodegen.d]: @ibuclaw why did you remove those aliases in #8591?

Comment (thewilsonator) [reply src/dmd/astcodegen.d]: They weren't, they were reordered alphabetically.

Comment (ZombineDev) [reply src/dmd/astcodegen.d]: Right, nevermind.
